test/provider_os_memory_config.cpp

```diff
@@ -119,21 +125,112 @@ TEST_F(providerConfigTest, protection_flag_read_write) {
+
+    res = umfOsMemoryProviderParamsSetPartSize(nullptr, 0);
+    ASSERT_EQ(res, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+
+    res = umfOsMemoryProviderParamsSetPartitions(nullptr, nullptr, 0);
+    ASSERT_EQ(res, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+}
+
+TEST_F(providerConfigTest, set_params_shm_name) {
+    umf_result_t res = umfOsMemoryProviderParamsSetShmName(params, nullptr);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    res = umfOsMemoryProviderParamsSetShmName(params, "shm_name");
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    res = umfOsMemoryProviderParamsSetShmName(params, "");
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    res = umfOsMemoryProviderParamsSetShmName(params, nullptr);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+}
+
+TEST_F(providerConfigTest, set_params_numa_list) {
+    unsigned numa_list[1] = {0};
+
+    umf_result_t res = umfOsMemoryProviderParamsSetNumaList(params, nullptr, 0);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    res = umfOsMemoryProviderParamsSetNumaList(params, numa_list, 1);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    res = umfOsMemoryProviderParamsSetNumaList(params, nullptr, 1);
+    ASSERT_EQ(res, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+
+    res = umfOsMemoryProviderParamsSetNumaList(params, numa_list, 0);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    // repeat the valid set to check memory leaks under Valgrind
+    res = umfOsMemoryProviderParamsSetNumaList(params, numa_list, 1);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+}
+
+TEST_F(providerConfigTest, set_params_partitions) {
+    umf_numa_split_partition_t partitions[1] = {{0, 1}};
+
+    umf_result_t res =
+        umfOsMemoryProviderParamsSetPartitions(params, nullptr, 0);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    res = umfOsMemoryProviderParamsSetPartitions(params, partitions, 1);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    res = umfOsMemoryProviderParamsSetPartitions(params, nullptr, 1);
+    ASSERT_EQ(res, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+
+    res = umfOsMemoryProviderParamsSetPartitions(params, partitions, 0);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+
+    // repeat the valid set to check memory leaks under Valgrind
+    res = umfOsMemoryProviderParamsSetPartitions(params, partitions, 1);
+    ASSERT_EQ(res, UMF_RESULT_SUCCESS);
+}
+
 struct providerConfigTestNumaMode
     : providerConfigTest,
       testing::WithParamInterface<umf_numa_mode_t> {
     struct bitmask *allowed_nodes = nullptr;
-    umf_os_memory_provider_params_t params = umfOsMemoryProviderParamsDefault();
+    umf_numa_mode_t expected_numa_mode;
+    //umf_os_memory_provider_params_handle_t params = nullptr;
 
     void SetUp() override {
         providerConfigTest::SetUp();
-        params.numa_mode = GetParam();
+
+        expected_numa_mode = GetParam();
+
+        auto res =
+            umfOsMemoryProviderParamsSetNumaMode(params, expected_numa_mode);
+        ASSERT_EQ(res, UMF_RESULT_SUCCESS);
```

Comment:
The `umf_test-provider_os_memory_config` test fails in this place on WSL (while on the main branch it does not):
```
[ RUN      ] numa_modes/providerConfigTestNumaMode.numa_modes/3
/home/ldorau/work/unified-memory-framework/test/provider_os_memory_config.cpp:30: Skipped
Test skipped, NUMA not available

[PID:17534  TID:17534  ERROR UMF] umfOsMemoryProviderParamsSetNumaMode: OS memory provider params handle is NULL
/home/ldorau/work/unified-memory-framework/test/provider_os_memory_config.cpp:226: Failure
Expected equality of these values:
  res
    Which is: 3
  UMF_RESULT_SUCCESS
    Which is: 0

[  FAILED  ] numa_modes/providerConfigTestNumaMode.numa_modes/3, where GetParam() = 5 (0 ms)
[----------] 4 tests from numa_modes/providerConfigTestNumaMode (1 ms total)

[----------] Global test environment tear-down
[==========] 12 tests from 2 test suites ran. (2 ms total)
[  PASSED  ] 0 tests.
[  SKIPPED ] 8 tests, listed below:
[  SKIPPED ] providerConfigTest.protection_flag_none
[  SKIPPED ] providerConfigTest.protection_flag_read
[  SKIPPED ] providerConfigTest.protection_flag_write
[  SKIPPED ] providerConfigTest.protection_flag_read_write
[  SKIPPED ] providerConfigTest.set_params_null_params_handle
[  SKIPPED ] providerConfigTest.set_params_shm_name
[  SKIPPED ] providerConfigTest.set_params_numa_list
[  SKIPPED ] providerConfigTest.set_params_partitions
[  FAILED  ] 4 tests, listed below:
[  FAILED  ] numa_modes/providerConfigTestNumaMode.numa_modes/0, where GetParam() = 0
[  FAILED  ] numa_modes/providerConfigTestNumaMode.numa_modes/1, where GetParam() = 1
[  FAILED  ] numa_modes/providerConfigTestNumaMode.numa_modes/2, where GetParam() = 2
[  FAILED  ] numa_modes/providerConfigTestNumaMode.numa_modes/3, where GetParam() = 5

 4 FAILED TESTS
```

Reply:
The reason is that `providerConfigTest::SetUp()` method calls `numa_available()` and it returns error and the `GTEST_SKIP` is called. So the `providerConfigTest` is skipped.
But the `GTEST_SKIP` does not affect the derived `providerConfigTestNumaMode` class. So the issue existed even before this PR, but with this PR we see a crash because the `params` structure was not created. BUt before this PR I do not understand how the `providerConfigTestNumaMode` worked without `libnuma`.

Do you have any idea how to fix that?

Reply:
I do not.
@lukaszstolarczuk do you have any idea?

Reply:
Maybe this commit 39cdf856b788d0ec84a86e18ea8d037011c75762 may help?

Reply:
Ok, looks like adding the following check helps:
```c
if (::providerConfigTest::IsSkipped()) {
    GTEST_SKIP();
}
```

Reply:
OK, it works now.